### DISCUSSION TITLE
Unable to find component: [profile.update-profile-information-form]

### DISCUSSION THREAD

**Discussion by tvarwig** [livewire/livewire#1764] [1. Help]
### Description
When I go to the default `/user/profile` route I get `Unable to find component: [profile.update-profile-information-form]` I can load the livewire views that I create but none of them are found in the default vendor folder.  I deleted my vendor folder and ran `composer upgrade` got a fresh vendor folder and I am the getting same issue. 

### Exact steps to reproduce
run `composer upgrade`
run `php artisan view:clear`
run `php artisan livewire:discover`
go to `/user/profile`
get error
### Stripped-down, copy-pastable code snippets
```
<div>
            <div class="max-w-7xl mx-auto py-8 sm:px-6 lg:px-8">

               @if(!Auth::user()->active)
               <div class="mb-6">
                   @livewire('account-not-active-alert', ['color' => 'yellow'])
               </div>
                @endif
                @livewire('profile.update-profile-information-form')

                @if (Laravel\Fortify\Features::enabled(Laravel\Fortify\Features::updatePasswords()))
                    <x-jet-section-border />

                    <div class="mt-10 sm:mt-0">
                        @livewire('profile.update-password-form')
                    </div>
                @endif

                @if (Laravel\Fortify\Features::canManageTwoFactorAuthentication())
                    <x-jet-section-border />

                    <div class="mt-10 sm:mt-0">
                        @livewire('profile.two-factor-authentication-form')
                    </div>
                @endif

                <x-jet-section-border />

                <div class="mt-10 sm:mt-0">
                    @livewire('profile.logout-other-browser-sessions-form')
                </div>

                <x-jet-section-border />

                <div class="mt-10 sm:mt-0">
                    @livewire('profile.delete-user-form')
                </div>
            </div>
        </div>
```
**Context**
- Livewire version: ^2.2.9
- Laravel version: ^8.0
- Browser: Chrome

**Comment by austenc** [comment]
@tvarwig could you share a github repo to make this easier to reproduce? There's unfortunately not enough context with the code you pasted here.

**Reply by sujilkumarkm** [reply]
Livewire\Exceptions\ComponentNotFoundException
Unable to find component: [profile.update-more-profile-form]
http://127.0.0.1:8000/user/profiles


I am gettting similar Error with Laravel 8 Jetstream - Fortify
![Capture](https://user-images.githubusercontent.com/64840932/97305155-2e7bb980-1876-11eb-802d-348cff8c0df1.PNG)

**Comment by sathia-musso** [comment]
I have the exact same issue, everything was working fine and then suddenly I receive this error. I tried clearing all caches and even reverting to previous commits. still no go. I wonder what it can be

**Comment by evently-nl** [comment]
Not sure if it helps, and I'm not sure why, but I got this error when installing doc-dress, did not have the time to look into it, but removing that package solved it for me. So could be this/another package that you installed?

**Reply by sathia-musso** [reply]
you may be onto something actually, I removed a package and stuff started working again. I guess there are some incompatibilities then. fyi the package was `livewire-calendar`, fun stuff

**Reply by iohaneiro** [reply]
I had the same problem with the package  asantibanez/livewire-calendar, I removed it, and it works fine again

**Comment by almassar** [comment]
I removed the infureal / artisan-gui package, after that it worked. There may be incompatibility with other packages

**Reply by RhysLees** [reply]
This worked for me thanks

**Comment by Flobbos** [comment]
Hello, 

I may be late to the party but I have the same issue with blade-ui-kit/blade-icons. This is highly frustrating because the problem is highly unpredictable. It works locally, but not on the server or the other way around.

**Reply by mjsoltani** [reply]
can you solve it?

**Comment by Said20hr** [comment]
I had the same issue, I solved this problem by deleting the existing livewire component and creating another one by the command make livewire [component name ]

**Comment by shopapps** [comment]
Not sure why this happened to me as well, I'm guessing there is some kind of Jetsream re-register command that could be called?

but for me I had already had a `App\Providers \LivewireServiceProvider` class that I used for registering specific components so I was able to copy the relevant contents of `Laravel\Jetstream\JetstreamServiceProvider` into that.

example file:
```
<?php

namespace App\Providers;

use Illuminate\Support\ServiceProvider;
use Illuminate\Support\Str;
use Laravel\Jetstream\Features;
use Laravel\Jetstream\Http\Livewire\ApiTokenManager;
use Laravel\Jetstream\Http\Livewire\CreateTeamForm;
use Laravel\Jetstream\Http\Livewire\DeleteTeamForm;
use Laravel\Jetstream\Http\Livewire\DeleteUserForm;
use Laravel\Jetstream\Http\Livewire\LogoutOtherBrowserSessionsForm;
use Laravel\Jetstream\Http\Livewire\NavigationMenu;
use Laravel\Jetstream\Http\Livewire\TeamMemberManager;
use Laravel\Jetstream\Http\Livewire\TwoFactorAuthenticationForm;
use Laravel\Jetstream\Http\Livewire\UpdatePasswordForm;
use Laravel\Jetstream\Http\Livewire\UpdateProfileInformationForm;
use Laravel\Jetstream\Http\Livewire\UpdateTeamNameForm;
use Livewire\Livewire;

class LivewireServiceProvider extends ServiceProvider
{
    protected array $livewireComponents = [
        'cancellation-journey'      => CancellationJourney::class, // <-- you can remove these, they are for demo purposes
        'cancellation-step-welcome' => Welcome::class,
        ActiveOrderCount::class,
    ];

    public function register() : void
    {

    }
    public function boot() : void
    {

        $this->registerLivewireComponents();
    }
    protected function registerLivewireComponents() : void
    {
        foreach ($this->livewireComponents as $alias => $class) {

            if(is_numeric($alias)){
                $reflection_class = new \ReflectionClass($class);
                $namespace = $reflection_class->getNamespaceName();
                $class_name = $reflection_class->getShortName();
                $alias = Str::of($class_name)
                    ->after($namespace . '\\')
                    ->replace(['/', '\\'], '.')
                    ->explode('.')
                    ->map([Str::class, 'kebab'])
                    ->implode('.');
            }
            Livewire::component($alias, $class);
        }

/*
 * Re-register the jetstream components 
 */

        Livewire::component('navigation-menu', NavigationMenu::class);
        Livewire::component('profile.update-profile-information-form', UpdateProfileInformationForm::class);
        Livewire::component('profile.update-password-form', UpdatePasswordForm::class);
        Livewire::component('profile.two-factor-authentication-form', TwoFactorAuthenticationForm::class);
        Livewire::component('profile.logout-other-browser-sessions-form', LogoutOtherBrowserSessionsForm::class);
        Livewire::component('profile.delete-user-form', DeleteUserForm::class);

        if (Features::hasApiFeatures()) {
            Livewire::component('api.api-token-manager', ApiTokenManager::class);
        }

        if (Features::hasTeamFeatures()) {
            Livewire::component('teams.create-team-form', CreateTeamForm::class);
            Livewire::component('teams.update-team-name-form', UpdateTeamNameForm::class);
            Livewire::component('teams.team-member-manager', TeamMemberManager::class);
            Livewire::component('teams.delete-team-form', DeleteTeamForm::class);
        }
    }
}
```

**Comment by MusheAbdulHakim** [comment]
In Laravel 10, jetstream 3.3,
Go to JetstreamSErviceProvider register method and add the form along with the respective FormClasses. Like: 
```
public function register(): void
    {
      $this->callAfterResolving(BladeCompiler::class, function () {
          if (config('jetstream.stack') === 'livewire' && class_exists(Livewire::class)) {
              Livewire::component('profile.update-profile-information-form', UpdateProfileInformationForm::class);
              Livewire::component('profile.update-password-form', UpdatePasswordForm::class);
              Livewire::component('profile.two-factor-authentication-form', TwoFactorAuthenticationForm::class);
              Livewire::component('profile.logout-other-browser-sessions-form', LogoutOtherBrowserSessionsForm::class);
              Livewire::component('profile.delete-user-form', DeleteUserForm::class);
          }
      });
    }
```

**Reply by amralsaleeh** [reply]
I tried it out but still didn't work!

**Reply by MusheAbdulHakim** [reply]
what's your laravel and jetstream version?.
You could also try reinstalling jetstream or running composer update and clearing cache

**Reply by amralsaleeh** [reply]
Laravel: 10.33.0
Jetstream: 4.0.2

I tried to run the composer update and clear the cache but neither worked

**Reply by MusheAbdulHakim** [reply]
ok. I am using jetstream 3.3 so that's probably why it's not working

**Comment by joleenshook** [comment]
upgraded to version 4.1.0 and it works fine.  Was broken on 4.03 for me.

**Comment by DevIP98** [comment]
It happened to me too "Property [$id] not found on component: [profile.update-profile-information-form]" and I solved it with:

` php artisan view:clear`

**Comment by Kenneth-Inkum** [comment]
So, I see this issue is finding its way into projects in different ways. For me, it's using Livewire Volt in Jetstream. Everything done to the T; just like in the documentation but I can't seem to fix the issue of "Unable to find component: [component-name]"

**Reply by amralsaleeh** [reply]
I faced this problem in my project for more than two months. I tried all the solutions mentioned above, but nothing worked for me.

However, I recently tried running:

```
composer update
```

And everything ran smoothly again.

I believe one of my project packages was causing the issue, but I didn't know which one.

Hope that helps! 💪

**Comment by juniorsanca** [comment]
Hello friends,

Follow this -> https://laravel-livewire.com/docs/2.x/installation

and everything will work fine !

dont forget to -> 
php artisan view:clear
php artisan config:clear
php artisan cache:clear
php artisan route:clear

Hope that helps :)
